### PR TITLE
Add the option to have a dashed border around a bar

### LINKED ISSUE #1144: Add feature to add dashed border around a bar
Please enable to add a feature to add dashed border to a bar in the barchart. This can useful when the user need to present a predicted data in the future. Below is a sample graph that I took from an app.

3-mint-ios (2)

### PR DESCRIPTION
**Pull request by k0psutin**
Add optional `borderDashArray` to `BarChartRodData`  and utilize `toDashedPath` with bar border.

Web:
Näyttökuva 2023-10-16 183207

Android:
Screenshot_1697470672

Closes #1144

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+## 0.65.0
```
Comment (imaNNeo) [inline CHANGELOG.md:1]: You don't need to specify a version here, just write `## newVersion` and I will bump a version when I decided to release.
Please read the [contributing guideline](https://github.com/imaNNeo/fl_chart#contributing) carefully

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,6 @@
...
+* **FEATURE** (by @k0psutin) Add dashed border to BarChartRodData, #1144
```
Comment (imaNNeo) [inline CHANGELOG.md:2]: #1114 is mentioned wrongly

File: lib/src/chart/bar_chart/bar_chart_data.dart [hunk 3]
```diff
@@ -353,6 +355,9 @@ class BarChartRodData with EquatableMixin {
   /// If you want to have a rounded rod, set this value.
   final BorderRadius? borderRadius;
 
+  /// If you want to have dashed border, set this value.
+  final List<int>? dashArray;
```
Comment (imaNNeo) [inline lib/src/chart/bar_chart/bar_chart_data.dart:359]: I think it makes more sense if we use `borderDashArray` here. What do you think?

**Comment by imaNNeo** [comment]
This PR is approved!
Please fix the discussions then we're ready for merge.

File: lib/src/chart/bar_chart/bar_chart_data.dart (hunk 3 above)
Comment (k0psutin) [reply lib/src/chart/bar_chart/bar_chart_data.dart:359]: Ah true, dashArray would be a bit confusing on its own.

File: CHANGELOG.md (hunk 1 above)
Comment (k0psutin) [reply CHANGELOG.md]: Merged with master and it is now under `##nextVersion`

File: lib/src/chart/bar_chart/bar_chart_data.dart (hunk 3 above)
Comment (k0psutin) [reply lib/src/chart/bar_chart/bar_chart_data.dart]: Renamed to borderDashArray, resolving conversation

**Comment by imaNNeo** [comment]
Bro, you should not merge the master into your branch, you need to `rebase` your branch. (read more about it [here](https://github.com/imaNNeo/fl_chart/blob/master/CONTRIBUTING.md#keep-your-branch-updated)).

I think you need to recreate your branch.
Everything looks green, but merge button is disabled.

image

**Comment by k0psutin** [comment]
Ah sorry about that. I reverted the merge and rebased. Hopefully it works now.

**Comment by k0psutin** [comment]
The tests _should_ be okay - I couldn't run the golden tests because of the pixel test failed. I mentioned about this at #1468. I ran the other tests separate from the golden files and they were ok.

**Comment by imaNNeo** [comment]
You're right, I just removed the golden tests for now https://github.com/imaNNeo/fl_chart/commit/0ae88cff737e226a837316b8547b33163552cfd0. I need to learn more about them in the future.
Please rebase one more time

**Comment by k0psutin** [comment]
Now it is rebased again and fixed the test to use TextScaler

**Comment by imaNNeo** [comment]
Thanks for your contribution!

**Pull request merged**